hack/compare-k8s-feature-gates.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+#
+# SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+usage() {
+  echo "Usage:"
+  echo "> compare-k8s-feature-gates.sh [ -h | <old version> <new version> ]"
+  echo
+  echo ">> For example: compare-k8s-feature-gates.sh 1.26 1.27"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+versions=("$1" "$2")
+out_dir=$(mktemp -d)
+function cleanup_output {
+    rm -rf "$out_dir"
+}
+trap cleanup_output EXIT
+
+for version in "${versions[@]}"; do
+  versions_dir=test/compatibility_lifecycle/reference
+  
+  # TODO Drop this when Kubernetes v1.34 gets released
+  if [ "$version" == "1.32" ]; then
+    versions_dir="test/featuregates_linter/test_data"
+  fi
```

Comment:
1. Why we have to drop the handling for K8s 1.32 when K8s 1.34 gets released. Theoretically, we want such script to work for any available K8s versions and not only for the K8s versions that we are about to add support for.

2. From https://github.com/gardener/gardener/blob/v1.118.2/docs/development/process.md#todo-statements:
> Each TODO statement should have an associated person and state when it can be removed.

I don't see associated person with this TODO. Anyways, let's remove this TODO if possible so that the script can support also old K8s versions.